presto-spi/src/main/java/io/prestosql/spi/connector/UpdateKind.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+public enum UpdateKind
```

Comment:
Why is this in the SPI? It's only used as an internal detail of the analyzer.

Reply:
It's a leftover from an earlier era of the UPDATE API.  It used to be an argument to `ConnectorMetadata.getUpdateRowIdColumnHandle()` before we broke that into two different methods, one for DELETE and one for UPDATE.  It's a great point that it doesn't belong in the SPI.  Demoted to be a private enum in `StatementAnalyzer`.